mempool/src/shared_mempool.rs

```diff
@@ -96,51 +97,68 @@ where
     subscribers: Vec<UnboundedSender<SharedMempoolNotification>>,
 }
 
-/// Message sent from Consensus to Mempool
-pub enum MempoolRequest {
-    /// get block req
-    GetBlockRequest {
-        /// max block size
-        max_block_size: u64,
-        /// txns in the block
-        transactions: Vec<TransactionExclusion>,
-    },
-    /// commit txns req
-    CommitTransactionsRequest {
-        /// txns to commit
-        transactions: Vec<CommittedTransaction>,
-        /// timestamp of block
-        block_timestamp_usecs: u64,
-    },
+/// Message sent from consensus to mempool
+pub enum ConsensusRequest {
+    /// request to pull block to submit to consensus
+    GetBlockRequest(
+        // max block size
+        u64,
+        // transactions to exclude from requested block
+        Vec<TransactionExclusion>,
+        // callback to send response back to sender
+        oneshot::Sender<Result<ConsensusResponse>>,
+    ),
+    /// notifications about *rejected* committed txns
+    CommitNotification(
```

Comment:
nit: if it's only for rejected transactions, can we rename it to RejectNotification or TransactionFailure